node/impl/full/eth.go

```diff
@@ -570,12 +571,78 @@ func (a *EthModule) EthGetTransactionReceiptLimited(ctx context.Context, txHash
 	return &receipt, nil
 }
 
-func (a *EthAPI) EthGetTransactionByBlockHashAndIndex(context.Context, ethtypes.EthHash, ethtypes.EthUint64) (ethtypes.EthTx, error) {
-	return ethtypes.EthTx{}, ErrUnsupported
+func (a *EthAPI) EthGetTransactionByBlockHashAndIndex(ctx context.Context, blkHash ethtypes.EthHash, index ethtypes.EthUint64) (*ethtypes.EthTx, error) {
+	ts, err := a.Chain.GetTipSetByCid(ctx, blkHash.ToCid())
+	if err != nil {
+		return nil, xerrors.Errorf("block not found")
+	}
+
+	return a.getTransactionByTipsetAndIndex(ctx, ts, index)
 }
 
-func (a *EthAPI) EthGetTransactionByBlockNumberAndIndex(context.Context, ethtypes.EthUint64, ethtypes.EthUint64) (ethtypes.EthTx, error) {
-	return ethtypes.EthTx{}, ErrUnsupported
+func (a *EthAPI) EthGetTransactionByBlockNumberAndIndex(ctx context.Context, blkParam string, index ethtypes.EthUint64) (*ethtypes.EthTx, error) {
+	validBlockTags := map[string]bool{
+		"latest":   true,
+		"pending":  true,
+		"earliest": true,
+		"final":    true,
+	}
+
+	var blockNumber string
+
+	if validBlockTags[blkParam] {
+		blockNumber = blkParam
+	} else {
+		blkNum, err := ethtypes.EthUint64FromHex(blkParam)
+		if err != nil {
+			return nil, fmt.Errorf("invalid block number format: %w", err)
+		}
+		blockNumber = strconv.FormatUint(uint64(blkNum), 10)
+	}
```

Comment:
```suggestion
	var blockNumber string

	switch blkParam {
	case "latest", "pending", "earliest", "final":
		blockNumber = blkParam
	default:
		blkNum, err := ethtypes.EthUint64FromHex(blkParam)
		if err != nil {
			return nil, xerrors.Errorf("invalid block number format: %w", err)
		}
		blockNumber = strconv.FormatUint(uint64(blkNum), 10)
	}
```

also note the use of `xerrors` here - it's used in lotus so we get error chaining, which `fmt.Errorf` doesn't support

Reply:
I'm not sure we need all this checking anyway, doesn't `getTipsetByBlockNumber` do it for us? wouldn't it just be the same as EthGetBlockByNumber?

Reply:
I think `getTipsetByBlockNumber ` should just work.